src/parser/metrics/Complexity.ts

```diff
@@ -1,55 +1,165 @@
+                } else {
+                    haveNoDefaultNodeType.push(languageAbbr);
+                }
+            }
+
+            // For languages which have also a default node type, always count this case node type:
+            if (haveDefaultNodeType.length > 0) {
+                this.complexityStatementsSuperSet.push(
+                    new ExpressionQueryStatement(
+                        caseNodeType.expression,
+                        haveDefaultNodeType,
+                        caseNodeType.activated_for_languages
+                    )
+                );
+            }
+
+            // Special query for languages which have no explicit default node type:
+            if (haveNoDefaultNodeType.length > 0) {
+                this.addCaseDefaultDifferentiatingQuery(haveNoDefaultNodeType, caseNodeType);
+            }
+        }
+    }
+
+    /**
+     * Adds a query statement that differentiates if the syntax node type is used as case or default label.
+     * For node types which are used for both case labels and default labels.
+     * @param nonDefaultLangAbbrs Abbreviations of the languages which use the node type as case label and
+     * have no explicit default label.
+     * @param caseDefaultNodeType The syntax node type.
+     */
+    addCaseDefaultDifferentiatingQuery(
+        noDefaultLangAbbrs: string[],
+        caseDefaultNodeType: ExpressionMetricMapping
+    ) {
+        if (caseDefaultNodeType.expression == "case_statement") {
+            // Special treatment for "case_statement" used by at least C++ and PHP.
+            // This syntax node can have more than one child,
+            // because it also has the content of the case block as child(s).
+            //
+            // In this case, a case label can be differentiated from a default label by
+            // checking if there is a child with the field name "value":
+            this.complexityStatementsSuperSet.push(
+                new SimpleLanguageSpecificQueryStatement(
+                    "(case_statement value: _ ) @case_statement",
+                    noDefaultLangAbbrs,
+                    caseDefaultNodeType.activated_for_languages
+                )
+            );
+        } else if (caseDefaultNodeType.expression == "when_entry") {
+            // Special treatment for the "when_entry" used by Kotlin. Can also have more than one child.
+            //
+            // A conditional when_entry can be differentiated from an else when_entry by checking
+            // if the first child ist a "when_condition":
+            this.complexityStatementsSuperSet.push(
+                new SimpleLanguageSpecificQueryStatement(
+                    "(when_entry (when_condition)) @when_entry",
+                    noDefaultLangAbbrs,
+                    caseDefaultNodeType.activated_for_languages
+                )
+            );
+        } else {
+            // Standard treatment: check if the case label node has a named child:
```

Comment:
It is hard to see that you try to identify default labels in this else block. I am not super happy to have so much comments, but we should at least extend the comment to make that clear.

Reply:
The comments primarily serve the purpose to document what the speciality of the syntax node type is that is handled here, they are not really meant to explain how the code works.

However, I admit that I may overdo when it comes to wordy documentation. In this case, however, the others told me that they liked the comments here, too, so I suggest to keep them.
But I have nothing against making my comments even longer :D